wandb/sdk/artifacts/artifact.py

```diff
@@ -391,6 +391,16 @@ def version(self) -> str:
         assert self._version is not None
         return self._version
 
+    @property
+    def collection(self) -> ArtifactCollection:
+        """The collection this artifact came from, ."""
+        if self._state == ArtifactState.PENDING:
+            raise ArtifactNotLoggedError(self, "collection")
+        base_name = self.name.split(":")[0]
+        return ArtifactCollection(
```

Comment:
Just confirming here: So since we access the artifact through a collection, it doesn't matter that it might have been linked to multiple collections, right? This method will always return the collection through which the artifact was accessed?

Reply:
Correct. `source_name` etc. would get the sequence